pkg/etcdenvvar/etcd_env.go

```diff
@@ -283,6 +287,39 @@ func getUnsupportedArch(_ envVarContext) (map[string]string, error) {
 	}, nil
 }
 
+func getObservedTLSMinVersion(envVarContext envVarContext) (tlsutil.TLSVersion, error) {
+	var observedConfig map[string]interface{}
+	if err := yaml.Unmarshal(envVarContext.spec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return "", fmt.Errorf("failed to unmarshal the observedConfig: %w", err)
+	}
+	observedMinTLSVersion, _, err := unstructured.NestedString(observedConfig, "servingInfo", "minTLSVersion")
+	if err != nil {
+		return "", fmt.Errorf("couldn't get minTLSVersion from observedConfig: %w", err)
+	}
+
+	// map tls version to string recognized by etcd
+	v, err := crypto.TLSVersion(observedMinTLSVersion)
+	if err != nil {
+		return "", fmt.Errorf("couldn't get minTLSVersion from observedConfig: %w", err)
+	}
+	switch v {
+	case tls.VersionTLS13:
+		return tlsutil.TLSVersion13, nil
+	default:
+		return tlsutil.TLSVersion12, nil
+	}
```

Comment:
this already exists as a dedicated function in https://github.com/etcd-io/etcd/blob/19aa0dbe8fd6317a237bae9b6ea52a4f1b445b19/client/pkg/tlsutil/versions.go#L32-L47

Reply:
That does not seem to help me here. I'm going from a library-go `TLSVersion1x` string to an etcd `TLS1.x` string.

Reply:
oh I see, it's not the integer- sorry!